src/bp/core/services/admin/community.ts

```diff
@@ -131,12 +141,7 @@ export class CommunityAdminService implements AdminService {
     }
 
     await this.knex(this.botsTable).insert(bot)
-
-    botTemplate
-      ? await this.botConfigWriter.createFromTemplate(bot, botTemplate)
-      : await this.botConfigWriter.createEmptyBot(bot)
-
-    await this.botLoader.mountBot(bot.id)
+    await this.jobService.execJob('mount', () => this._addBot(bot, botTemplate))
```

Comment:
This won't work in a distributed environement as the job isn't define on the other nodes.  Also why name the job and not simply wrap it like `addBot = this.jobService.makeDistributed(this._addBot)`

Reply:
Ok so I think I'm starting to understand how this works. I can't pass a function around and expect it to be called on other nodes. I need to publish a message that will tell the other nodes what to do i.e. 'mount', 'unmount', etc. When the other nodes will pick up the message, they will execute the specific code associated with the message (some kind of switch case).

But what I don't understand is that you keep telling me to pass a function to the job service. I can't do that because like you said, only that node is aware of the function.

Reply:
In the meantime what I'm gonna do is to create delegates (onMount, onUnmount) in the jobService which will be implemented somewhere else. When I receive a 'mount' message, I'll call that delegate function.